activejob/lib/active_job/queue_adapters/shoryuken_adapter.rb

```diff
@@ -0,0 +1,67 @@
+require 'shoryuken'
+
+module ActiveJob
+  module QueueAdapters
+    # == Shoryuken adapter for Active Job
+    #
+    # Shoryuken ("sho-ryu-ken") is a super-efficient AWS SQS thread based message processor.
+    #
+    # Read more about Shoryuken {here}[https://github.com/phstc/shoryuken].
+    #
+    # To use Shoryuken set the queue_adapter config to +:shoryuken+.
+    #
+    #   Rails.application.config.active_job.queue_adapter = :shoryuken
+    class ShoryukenAdapter
+      class << self
+        def enqueue(job) #:nodoc:
+          register_worker!(job)
+
+          queue = Shoryuken::Client.queues(job.queue_name)
+          queue.send_message(message(job))
+        end
+
+        def enqueue_at(job, timestamp) #:nodoc:
+          register_worker!(job)
+
+          delay = (timestamp - Time.current.to_f).round
+          raise 'The maximum allowed delay is 15 minutes' if delay > 15.minutes
```

Comment:
Also this should be documented like [this](https://github.com/rails/rails/blob/master/activejob/lib/active_job/queue_adapters.rb#L56).